modules/programs/aerc-accounts.nix

```diff
@@ -221,7 +221,9 @@ in {
     // account.aerc.extraAccounts;
 
   mkAccountConfig = name: account:
-    mapAttrNames (addAccountName name) account.aerc.extraConfig;
+    (mapAttrNames (addAccountName name)
+      (lib.attrsets.filterAttrs (k: v: k == "ui") account.aerc.extraConfig))
+    // lib.attrsets.filterAttrs (k: v: k != "ui") account.aerc.extraConfig;
```

Comment:
For future reference: use `lib.partition` to do this in a single pass.

Reply:
Actually I read too fast and thought this was a list. If you're just extracting the `ui` key from an attrset, you can just do something like this

```nix
builtins.removeAttrs account.aerc.extraConfig [ "ui" ] //
optionalAttrs (account.aerc.extraConfig?ui) {
  ${addAccountName name "ui"} = account.aerc.extraConfig.ui;
}
```

Reply:
or

```nix
mapAttrNames (n: if n == "ui" then addAccountName name n else n) account.aerc.extraConfig
```